src/helpers.rs

```diff
@@ -118,6 +119,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.memory.write_bytes(ptr, data.iter().copied())
     }
 
+    /// Call a function: Push the stack frame and pass the arguments.
+    /// For now, arguments must be scalars (so that the caller does not have to know the layout).
+    fn call_function(
+        &mut self,
+        f: ty::Instance<'tcx>,
+        args: &[Scalar<Tag>],
+        dest: Option<PlaceTy<'tcx, Tag>>,
+        stack_pop: StackPopCleanup,
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        // Push frame.
+        let mir = this.load_mir(f.def, None)?;
+        this.push_stack_frame(
+            f,
+            DUMMY_SP, // There is no call site.
```

Comment:
That's not true in general. You can take the current frame's current statement/terminator and get a span from that. If there are no frames, only then we definitely can't have a span because nothing calls the start function

Reply:
As far as I can see, none of the code I am replacing did that correctly so far. They used `DUMMY_SP` or the span of the *callee* MIR, the latter being just blatantly wrong.

To avoid code duplication I'd prefer to add a `Frame::current_source_info` method on the rustc side and use that here. I'll add that to the rustc patch that I am working on anyway for https://github.com/rust-lang/miri/issues/1070.

Would you prefer to block this PR on that? Either way is fine for me.

Reply:
nah, this can totally happen later, you're right this is preexisting.